.github/workflows/Regression.yml

```
@@ -74,6 +85,18 @@ jobs:
           make
           cd ..
 
+      # For NightlyTest we fetch the last commit hash that successfully completed Regression on main
+      - name: Build Main and Previous Successful Regression Hash
+        if: ${{ github.repository == 'duckdb/duckdb' && github.ref == 'refs/heads/main' }}
+        shell: bash
+        run: |
+          make
+          git clone https://github.com/duckdb/duckdb.git
+          git checkout $(gh run list --repo duckdb/duckdb --branch=main --workflow=Regression --status=success --json=headSha --limit=1 --jq '.[0].headSha')
```

```suggestion
          cd duckdb && git checkout $(gh run list --repo duckdb/duckdb --branch=main --workflow=Regression --status=success --json=headSha --limit=1 --jq '.[0].headSha') && cd ..
```
Me again 👋, I think checkout needs to be done IN the duckdb folder.

---

I think adding this suggestion + merging is cool to me

---

Sorry I was not paying attention well, I think it should be good now